Microblogging/src/app/components/main-page/main-page.component.ts

```diff
@@ -67,6 +68,17 @@ export class MainPageComponent implements OnInit {
   }
 
   changeImage() {
-    console.log(this.url);
+    let obj: User = {
+      id: this.user.id,
+      username: this.user.username,
+      firstName: this.user.firstName,
+      lastName: this.user.lastName,
+      imageURL: this.imageURL,
```

Comment:
why don't you just set this.user.imageURL = this.imageURL and then pass this.user into updateUser?